client/my-sites/earn/memberships/add-edit-plan-modal.jsx

```diff
@@ -289,6 +298,13 @@ const RecurringPaymentsPlanAddEditModal = ( {
 						) }
 					/>
 				</FormFieldset>
+				<FormFieldset>
+					<ToggleControl
+						onChange={ handleMarkAsDonation }
+						checked={ 'donation' === editedMarkAsDonation }
+						label={ translate( 'Mark payment plan as a donation.' ) }
+					/>
+				</FormFieldset>
```

Comment:
I'm wondering if we should only allow folks to pick a product type when creating the plan, and disable the control when editing it. 🤔 
E.g.
```js
<ToggleControl disabled={ !! product } />
```

Like, there could be ethical (if not legal?) implications on changing an existing subscription from "donation" to "payment".
Disallowing editing the type would remove all our liabilities while keeping the system flexible enough.

What do you think?